Expensify/App
# [No QA] Cleanup implementation of HTMLEngineProvider

### Details
This PR cleans up the `HTMLEngineProvider` component implementation to make it align with the one-component-per-file convention we have throughout the rest of our codebase. There was a seemingly arbitrary exception made in `HTMLEngineProvider`, and it made the component very large and harder to scan/reason about.

### Fixed Issues
$ n/a

### Tests
1. Open a chat
1. Send a text-only message. Verify that it renders correctly.
1. Edit the text-only message. Verify that the edit box appears correctly and that you can edit the message.
1. Send an image attachment. Verify that it renders correctly.
1. Send an inline code block. Verify that it renders correctly.

### QA Steps
Just regression tests.

### Tested On


### Screenshots
#### Web
<img src="https://user-images.githubusercontent.com/47436092/148125799-e5b613f9-954c-47d9-9034-2f68c8391233.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148125843-92dc5efb-9084-420d-8c6d-5f971c8ba496.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148125873-4c30c53b-ed85-4215-8178-70593fa010fd.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148125971-58015df6-7cec-4a96-9092-a456be7403e9.png" alt="" width="350px" />

#### Mobile Web
<img src="https://user-images.githubusercontent.com/47436092/148126500-dfa3d815-272c-436a-8ec4-952a641386d4.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148126552-5e2a2d55-e7a4-44f7-aa0b-d37309f430d1.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148126616-b96b7b4a-e5bb-4f16-b50f-e351a020db25.png" alt="" width="350px" />

#### Desktop
<img src="https://user-images.githubusercontent.com/47436092/148126980-cbc6a1c5-dc60-4186-bc5d-a5b70d73dd19.png" alt="" width="350px" />

#### iOS
<img src="https://user-images.githubusercontent.com/47436092/148132237-c1395412-2f9d-4d3d-b225-1c55d2e65fec.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148132347-a7db75a8-1f19-4dde-af76-0088f067a43f.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148132424-62c33147-a092-4e44-9c8c-d0c0914b584b.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148132464-f11dd903-1e52-4b4a-920d-0b9d3a3aeadc.png" alt="" width="350px" />

#### Android
<img src="https://user-images.githubusercontent.com/47436092/148134706-e5e6c755-6c44-4393-a84d-b74c847ea7f6.png" alt="" width="350px" />
<img src="https://user-images.githubusercontent.com/47436092/148134763-97a060ed-8002-44f5-bed8-1f0f45a5fc84.png" alt="" width="350px" />
